staging/src/k8s.io/apiserver/pkg/storage/interfaces.go

```diff
@@ -251,6 +257,10 @@ type Interface interface {
 	// TODO: Remove when storage.Interface will be separate from etc3.store.
 	// Deprecated: Added temporarily to simplify exposing RequestProgress for watch cache.
 	RequestWatchProgress(ctx context.Context) error
+
+	// Supports checks whether the etcd instance supports the specified feature.
+	// The function accepts one Feature parameter representing the features to check.
+	Supports(feature Feature) (bool, error)
```

Comment:
I still don't really see how exactly this will be integrated into switching off feature/failing kube-apiserver based on not supporting the feature.

I would really like to see the e2e flow before merging this PR.

Reply:
Based on discussions on slack we decide on not disabling the feature but instead just log the user, we have two main functions for `etcdfeature` module, `Supports` and `ClientSupports`.
1. `Supports` just check the cache of endpoints supporting feature and return `true` if all features are supported. we use this on places without any etcd client.
2. `ClientSupports` computes and caches the support of each endpoint.

We use `ClientSupports` in etcd3/store.go in `Supports` method, and i think we can throw error or fail kube-apiserver from there.
Please correct me if i'm wrong or miss anything. @serathius

Reply:
Also please see `NewCacherFromConfig` we can fail api server there.